### PR TITLE
Fix Grape::Endpoint's inspect method when not called in the context of an API

### LINKED ISSUE #2491: Grape 2.1.x fail to compile, NoMethodError: undefined method `[]' for nil in grape/dsl/inside_route.rb:421:in `route'
I've an issue after upgrade Grape from 2.0 to 2.1.x 
My apps setup. The same error is produced on:
Ruby 3.3.4, Rails 7.2
Ruby 3.3.1 , Rails 7.0.8.4, 7.1.3.3
Official repo [Grape on Rails](https://github.com/ruby-grape/grape-on-rails) behaves the same after upgrade Grape locally to 2.1.3

Steps to reproduce:
- Use any of Ruby 3.x and Rails 7.x
- Use Twitter::API example form readme
- Try to compile `Twitter::API.compile!` with Grape 2.0
- Upgrade Grape to 2.1.*
- Try to compile `Twitter::API.compile!`

```
Twitter::Api.compile!
An error occurred when inspecting the object: #<NoMethodError: undefined method `[]' for nil>
An error occurred when running Kernel#inspect: #<NoMethodError: undefined method `[]' for nil>
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/grape-2.1.3/lib/grape/dsl/inside_route.rb:423:in `route'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/grape-2.1.3/lib/grape/endpoint.rb:409:in `inspect'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/activesupport-7.2.1/lib/active_support/ordered_options.rb:69:in `inspect'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/activesupport-7.2.1/lib/active_support/ordered_options.rb:69:in `inspect'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb/inspector.rb:101:in `inspect'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb/inspector.rb:101:in `inspect'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb/inspector.rb:101:in `inspect'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb/inspector.rb:101:in `inspect'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb/inspector.rb:101:in `bind_call'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb/inspector.rb:101:in `rescue in inspect_value'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb/inspector.rb:95:in `inspect_value'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb/context.rb:639:in `inspect_last_value'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1395:in `output_value'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1057:in `block (2 levels) in eval_input'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1388:in `signal_status'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1041:in `block in eval_input'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1120:in `block in each_top_level_statement'
<internal:kernel>:187:in `loop'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1117:in `each_top_level_statement'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1040:in `eval_input'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1021:in `block in run'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1020:in `catch'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/irb-1.14.0/lib/irb.rb:1020:in `run'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/commands/console/irb_console.rb:119:in `start'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/commands/console/console_command.rb:59:in `start'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/commands/console/console_command.rb:8:in `start'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/commands/console/console_command.rb:87:in `perform'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/thor-1.3.2/lib/thor/command.rb:28:in `run'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/thor-1.3.2/lib/thor/invocation.rb:127:in `invoke_command'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/command/base.rb:178:in `invoke_command'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/thor-1.3.2/lib/thor.rb:538:in `dispatch'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/command/base.rb:73:in `perform'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/command.rb:71:in `block in invoke'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/command.rb:149:in `with_argv'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/command.rb:69:in `invoke'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/railties-7.2.1/lib/rails/commands.rb:18:in `<main>'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
/Users/rails/.rbenv/versions/3.3.1/lib/ruby/gems/3.3.0/gems/bootsnap-1.18.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
bin/rails:4:in `<main>'
```

### PR DESCRIPTION
**Pull request by ericproulx**
Fix #2491

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 * [#2480](https://github.com/ruby-grape/grape/pull/2480): Fix rescue_from ValidationErrors exception - [@numbata](https://github.com/numbata).
 * [#2464](https://github.com/ruby-grape/grape/pull/2464): The `length` validator only takes effect for parameters with types that support `#length` method - [@OuYangJinTing](https://github.com/OuYangJinTing).
 * [#2485](https://github.com/ruby-grape/grape/pull/2485): Add `is:` param to length validator - [@dakad](https://github.com/dakad).
+* [#2492](https://github.com/ruby-grape/grape/pull/2492): Fix Grape::Endpoint's inspect method - [@ericproulx](https://github.com/ericproulx).
```
Comment (dblock) [inline CHANGELOG.md:16]:
```suggestion
* [#2492](https://github.com/ruby-grape/grape/pull/2492): Fix `Grape::Endpoint#inspect` - [@ericproulx](https://github.com/ericproulx).
```

File: lib/grape/endpoint.rb [hunk 2]
```diff
@@ -234,6 +234,15 @@ def equals?(endpoint)
       (options == endpoint.options) && (inheritable_setting.to_hash == endpoint.inheritable_setting.to_hash)
     end
 
+    # the purpose of this override is solely for stripping internals when an error occurs while calling
```
Comment (dblock) [inline lib/grape/endpoint.rb:237]:
```suggestion
    # The purpose of this override is solely for stripping internals when an error occurs while calling
```

File: lib/grape/endpoint.rb [hunk 3]
```diff
@@ -234,6 +234,15 @@ def equals?(endpoint)
...
+    # an endpoint through an api. See #https://github.com/ruby-grape/grape/issues/2398
```
Comment (dblock) [inline lib/grape/endpoint.rb:238]:
```suggestion
    # an endpoint through an api. See https://github.com/ruby-grape/grape/issues/2398.
```

File: lib/grape/endpoint.rb [hunk 4]
```diff
@@ -234,6 +234,15 @@ def equals?(endpoint)
...
+    # Otherwise, it calls super
```
Comment (dblock) [inline lib/grape/endpoint.rb:239]:
```suggestion
    # Otherwise, it calls super.
```

File: lib/grape/endpoint.rb [hunk 5]
```diff
@@ -234,6 +234,15 @@ def equals?(endpoint)
...
+    def inspect
+      return super unless env
+
+      "#{self.class} in `#{route.origin}' endpoint"
```
Comment (dblock) [inline lib/grape/endpoint.rb:243]: Was this meant to be quoted differently?

```suggestion
      "#{self.class} in '#{route.origin}' endpoint"
```

**Review by dblock — APPROVED** [review]
Some nits, thanks!

Is there a test one/you could write that gets us closer to the usage in the bug it fixes?

File: lib/grape/endpoint.rb (hunk 5 above)
Comment (ericproulx) [reply lib/grape/endpoint.rb:243]: https://bugs.ruby-lang.org/issues/16495
Looks like something cultural

**Comment by ericproulx** [comment]
> Some nits, thanks!
> 
> Is there a test one/you could write that gets us closer to the usage in the bug it fixes?

I've added a [test](https://github.com/ruby-grape/grape/blob/4f6a48daa11e107d18811f122960f7c20057c43e/spec/grape/endpoint_spec.rb) that calls `inspect` on a Endpoint instance not related to an API call.

**Review by dblock — APPROVED** [review]

**Pull request merged**